### PR TITLE
Implemented Shaking, Swinging, Tilting

### PR DESCRIPTION
**Pull request by Corentinthomee**
The ability to use Shaking, Swinging and Tilting from a python script have been added.
Let me know if anything is wrong, I'd be happy to fix it

### CODE REVIEW

**Comment by Felk** [comment]
Hey, thank you very much! On a first glance it looks good.

However, the problem is that for months now, or even close to a year, I've neglected finally syncing the fork with upstream. The problem is that the way input handling, and most notably programmatic input manipulation, is handled was refactored upstream. I've tried to fix it a couple of times now, didn't manage to and lost interest for months.

Now, if I merge this, the amount of stuff that needs to be refactored after updating and merging upstream increases. I'm kinda in a dilemma here. I'll get back to you when I know what I'm going to do with my fork here

**Comment by Felk** [comment]
I haven't gotten around to fix the python stubs yet, but I finally managed to port the controller code to the new input override system from upstream. Have a look at https://github.com/Felk/dolphin/commit/17f96f7310e78ffec9f94f438288aa72c4d16b2a

This is somewhat good and bad news in that it allowed me to easily add support for setting the accelerometer and angular velocity data, but in that move it also made your merge request obsolete. Before I'm closing this I'd like to get a bit of feedback though whether `controller.set_wii_acceleration(int, float, float, float)` (m/s²) and `controller.set_wii_angular_velocity(int, float, float, float)` (radians/s) is sufficiently useful or if it'd still be better to control Shake, Tilt and Swing individually. What do you think @Corentinthomee @VIPTankz ?

**Comment by Corentinthomee** [comment]
Hey,
I guess control over the accelerometer and gyroscope is technically enough, but it makes it complicated to simulate even simple motions. I still think the natural motions can be a plus.
The parts of the code that I modified did not seem to change much, so it shouldn't be too hard to re-implement the changes if needed. Let me know if I can do anything to help !

**Comment by Felk** [comment]
> I guess control over the accelerometer and gyroscope is technically enough, but it makes it complicated to simulate even simple motions. I still think the natural motions can be a plus.

Do I understand it correctly that the accelerometer and gyroscope are a more "raw" form of input, meaning they get translated into Shake, Swing and Tilt anyways? In that case I agree that being able to set those directly sounds useful

> The parts of the code that I modified did not seem to change much, so it shouldn't be too hard to re-implement the changes if needed.

Sure, I'd love for you to give it a try!

**Comment by Corentinthomee** [comment]
Hey, 
I managed to rewrite the changes for the new input system.
Everything seems to be working fine

**Review by Felk — CHANGES_REQUESTED** [review]
Thank you very much! The changes look very good and almost mergeable. I left a few comments.

File: python-stubs/dolphin/controller.pyi [hunk 1]
```diff
@@ -330,3 +330,78 @@ def set_gba_buttons(controller_id: int, inputs: GBAInputs, /) -> None:
...
+                      angle: float, velocity: float, /) -> None:
+    """
+    Sets the WiiMote tilt motion.
+
+    The override will hold for the current frame.
+    :param controller_id: 0-based index of the controller
+    :param x: x-component of the tilt
+    :param y: y-component of the tilt
+    :param angle: max tilt angle (deg/180)
+    :param velocity: rotational velocity of the tilt (rad/s)
```
[the remark this answers is not in the document]
Comment (Corentinthomee) [reply python-stubs/dolphin/controller.pyi:406]: The angle is later used as argument of the function SquareStickGate::SquareStickGate(ControlState half_width). This function is used at other places, with a default value of 1, so I guess a 180° rotation of the wiimote somehow relates to that. 
I used deg/180 in the controller function to be consistent with this, but it is really confusing. I'll convert the angle to radians to make it more usable.

**Review by Felk — APPROVED** [review]
Looks perfect, thank you very much!

**Pull request merged**